# Based on https://github.com/oracle/solaris-userland/blob/master/components/rsyslog/patches/

# This patch is a fix for proper uncompression
# source: written internally
# Should be sent upstream, but hasn't been.
# Bug has existed a long time (probably since rsyslog implemented compression),
# last noticed in 8.4.0
diff -wpruN '--exclude=*.orig' a~/runtime/parser.c a/runtime/parser.c
--- a~/runtime/parser.c	1970-01-01 00:00:00
+++ a/runtime/parser.c	1970-01-01 00:00:00
@@ -333,6 +333,15 @@ static rsRetVal uncompressMessage(smsg_t
 		 * feature.
 		 */
 		int ret;
+
+		/*
+		 * If NO_PRI_IN_RAW is set, this is a local message and
+		 * therefore not compressed (though it may well begin with
+		 * 'z'!).
+		 */
+		if ((pMsg->msgFlags & NO_PRI_IN_RAW) != 0)
+			FINALIZE;
+
 		iLenDefBuf = glbl.GetMaxLine();
 		CHKmalloc(deflateBuf = malloc(iLenDefBuf + 1));
 		ret = uncompress((uchar *) deflateBuf, &iLenDefBuf, (uchar *) pszMsg+1, lenMsg-1);
